Additional check in install.sh for RPM-based distros

Djelibeybi opened the pull request:
This resolves #14924 by adding an additional check for `dist_version` after all the version checking is done.

An alternative would be to move the `dist_version` check out of the initial checks and into the case block for each `lsb_dist` instead.

Signed-off-by: Avi Miller avi.miller@oracle.com

Commit:
Windows: CLI Improvement

The Ansi parser and their associated actions have been decoupled. Now
parsing results in call backs to an interface which performs the
appropriate actions depending on the environment.

This improvement provides a functional Vi experience and the vttest no
longer panics.

This PR replaces docker/docker #13224 with the latest console updates.

Signed-off-by: John Howard <jhoward@microsoft.com>

Commit:
Windows: Factoring out unused fields

Signed-off-by: John Howard <jhoward@microsoft.com>

Djelibeybi committed:
Additional check in install.sh for RPM-based distros to fix the dist_version possibly returned by lsb_release.

Signed-off-by: Avi Miller <avi.miller@oracle.com>

Djelibeybi committed:
Additional check in install.sh for RPM-based distros to fix the dist_version possibly returned by lsb_release.

Signed-off-by: Avi Miller <avi.miller@oracle.com>

thaJeztah commented:
/cc @tianon @jfrazelle for review

Commit:
Update vendored distribution repo to new version

This version includes a fix that avoids checking against specific HTTP
status codes. The previous behavior violated the registry API spec.

Fixes #14975

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

jfrazelle commented on hack/install.sh:
```diff
@@ -238,6 +238,8 @@ do_install() {
 
 		fedora|centos)
+			# The dist_version from lsb_release is wrong, so grab it again to be safe.
+			dist_version="$(rpm -q --whatprovides redhat-release --queryformat "%{VERSION}\n")"
```
but why? isnt this what your other PR is doing

Djelibeybi replied:
Nope. The other PR fixes the case when lsb_release *isn't* installed. This fixes the case when it is, because the logic to determine distro and version is skewed to Debian+derivatives. 

As I said in the PR, a possibly more elegant fix is to remove the dist_version check completely from the non-Debian distros and only check it later.

chrisseto committed:
Skip DockerSuite.TestExecResizeImmediatelyAfterExecStart on lxc

Signed-off-by: Chris Seto <chriskseto@gmail.com>
Signed-off-by: Avi Miller <avi.miller@oracle.com>

Commit:
Fix golint nit in term_windows.go

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

Djelibeybi committed:
Reworked the distro detection process to seperate the lsb_dist detection and dist_version detection.

This allows for more accurate detection of dist_version based on lsb_dist.

Tested on Fedora 22, Ubuntu 14.04, Debian 7/8, Oracle Linux 6/7, CentOS 7 and RHEL7.

Signed-off-by: Avi Miller <avi.miller@oracle.com>

Djelibeybi committed:
Reworked the distro detection process to seperate the lsb_dist detection and dist_version detection.

This allows for more accurate detection of dist_version based on lsb_dist.

Tested on Fedora 22, Ubuntu 14.04, Debian 7/8, Oracle Linux 6/7, CentOS 7 and RHEL7.

Signed-off-by: Avi Miller <avi.miller@oracle.com>

Djelibeybi commented:
Updated PR with reworked distro detection process, as discussed with @jfrazelle in IRC.

jfrazelle commented:
hmm looks like it needs a rebase sorry but thanks! for doing this, ill look it over!

jfrazelle commented on hack/install.sh:
```diff
@@ -237,7 +266,7 @@ do_install() {
 			;;
 
-		fedora|centos)
+		fedora|centos|oracleserver)
```
oracleserver wont work as lsb_dist for download in the url needs to be oraclelinux

Djelibeybi replied:
Except oracleserver is Oracle Linux's lsb_dist value. That's what is returned by lsb_release and what was assigned to us by LSB.

If you must have it as oraclelinux, then you'll need to split out oracleserver as the lsb_dist and replace lsb_dist with oraclelinux.

jfrazelle replied:
ya we kinda have tooo :(

mostly because it is generated in the rpm build from the dockerfiles

On Mon, Jul 27, 2015 at 8:04 PM, Avi Miller <notifications@github.com>
wrote:

> In hack/install.sh
> <https://github.com/docker/docker/pull/14927#discussion_r35610818>:
>
> > @@ -237,7 +266,7 @@ do_install() {
> >  			exit 0
> >  			;;
> >
> > -		fedora|centos)
> > +		fedora|centos|oracleserver)
>
> Except oracleserver is Oracle Linux's lsb_dist value. That's what is
> returned by lsb_release and what was assigned to us by LSB.
>
> If you must have it as oraclelinux, then you'll need to split out
> oracleserver as the lsb_dist and replace lsb_dist with oraclelinux.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/14927/files#r35610818>.
>

Djelibeybi replied:
Ok, I'll fix it up and rebase at the same time. Give me a few minutes.

jfrazelle replied:
no prob and no rush, its nighttimme haha

On Mon, Jul 27, 2015 at 8:10 PM, Avi Miller <notifications@github.com>
wrote:

> In hack/install.sh
> <https://github.com/docker/docker/pull/14927#discussion_r35611057>:
>
> > @@ -237,7 +266,7 @@ do_install() {
> >  			exit 0
> >  			;;
> >
> > -		fedora|centos)
> > +		fedora|centos|oracleserver)
>
> Ok, I'll fix it up and rebase at the same time. Give me a few minutes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/14927/files#r35611057>.
>

Djelibeybi committed:
Added specific check for Oracle Linux to rewrite the lsb_dist to the URL path
component.

Also removed checks for redhatenterpriseserver because this is replaced with
centos higher up anyway.

Signed-off-by: Avi Miller <avi.miller@oracle.com>

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "fix14924" git@github.com:Djelibeybi/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~9
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

Djelibeybi commented:
Hmm. I'm new to rebasing, so I think I may have stuffed this up. :( Sorry!

Djelibeybi commented:
This PR is broken. Will submit a new one.
